metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -413,7 +413,7 @@ object SemanticTokensProvider {
   object DocstringToken {
 
     val paramOrThrows: String = "(@param|@tparam|@throws)\\s+([\\w.]+)"
-    val apiLink: String = "\\[\\[(.*)\\]\\]"
+    val apiLink: String = "\\[\\[([^\\]]*)\\]\\]"
```

Comment:
In theory you can write:
```
[[example.O.bar[T]():Int]]
```
I'd use a non-greedy operator instead and do: `"\\[\\[(.*?)\\]\\]"`

Reply:
Right, thanks